pkg/route/api/v1/types.go

```diff
@@ -48,6 +48,13 @@ type RouteSpec struct {
 
 	// TLS provides the ability to configure certificates and termination for the route
 	TLS *TLSConfig `json:"tls,omitempty" description:"provides the ability to configure certificates and termination for the route"`
+
+	// RouterShard has information of a routing shard and is used to
+	// generate host names
+	Shard *RouterShard `json:"shard,omitempty" description:"provides shard information for the route"`
+
+	// If specfied and true then DNSSuffix is appended to display in UI when host is specified for a route
+	AppendDNSSuffix bool `json:"appendDNSSuffix,omitempty" description:"provides information whether to append DNSSuffix to display in UI when host is specified"`
```

Comment:
Right now, spec.host is fully qualified. to get different behavior requires an API change the router will honor, not just a flag for UI/display. I'll let @smarterclayton weigh in on how will a single flag fits with the direction of routes (this would not work with multiple routers, or multiple possible suffixes, etc)